pkg/controller/elasticsearch/filesettings/reconciler.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package filesettings
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	esv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+)
+
+// ReconcileEmptyFileSettingsSecret reconciles an empty File settings Secret for the given Elasticsearch only when there is no Secret.
+// Used by the Elasticsearch controller.
+func ReconcileEmptyFileSettingsSecret(
+	ctx context.Context,
+	c k8s.Client,
+	es esv1.Elasticsearch,
+	createOnly bool,
```

Comment:
At some point, `ReconcileEmptyFileSettingsSecret` was called in `resetOrphanSoftOwnedSecrets` in the SCP controller with `createOnly=false` but then I discovered that `ReconcileSecret` doesn't reset the annotation. `ReconcileSecret` is ok to use to create an empty Secret but not to update to an empty Secret where we want to also reset the annotations.

Instead of replacing `ReconcileSecret` by `ReconcileResource` in `ReconcileEmptyFileSettingsSecret`, I just rewrote
`resetOrphanSoftOwnedSecrets` to use `ReconcileResource`.

I now think I should use `ReconcileResource` in `ReconcileEmptyFileSettingsSecret` and calls it from `resetOrphanSoftOwnedSecrets`. Then I need the boolean:

```go
pkg/controller/elasticsearch/driver/driver.go
313:	err = filesettings.ReconcileEmptyFileSettingsSecret(ctx, d.Client, d.ES, true)

pkg/controller/stackconfigpolicy/controller.go
439:		err = filesettings.ReconcileEmptyFileSettingsSecret(ctx, c, es, false)
```